runtime/test/planner-tests.js

```diff
@@ -580,3 +580,59 @@ describe('MatchParticleByVerb', function() {
                      plans.map(plan => plan.particles.map(particle => particle.name)));
   });
 });
+describe('FallbackFate', function() {
+  it('fallback for search based recipe', async () => {
+    let manifest = (await Manifest.parse(`
+      schema Thing
+      particle DoSomething in 'AA.js'
+        DoSomething(in Thing inthing, out Thing outthing)
+
+      recipe
+        search \`DoSomething\`
+        use as view0
+        use as view1
+        DoSomething as particle0
+          inthing <- view0
+          outthing -> view1
+    `));
+    let recipe = manifest.recipes[0];
+    recipe.views.forEach(v => v._originalFate = "?");
+    assert(recipe.normalize());
+    var arc = createTestArc("test-plan-arc", manifest, "dom");
+    var strategizer = {generated: [{result: manifest.recipes[0], score: 1}], terminal: []};
+
+    var ff = new FallbackFate(arc);
+    let { results } = await ff.generate(strategizer);
+
+    assert.equal(results.length, 1);
+    let plan = results[0].result;
+    assert.equal(plan.views.length, 2);
+    assert.equal('map', plan.views[0].fate);
+    assert.equal('copy', plan.views[1].fate);
+  });
+
+  it('ignore non-search unresolved recipe', async () => {
+    // Same as above, but the recipe doesn't originate from user search query.
+    let manifest = (await Manifest.parse(`
+      schema Thing
+      particle DoSomething in 'AA.js'
+        DoSomething(in Thing inthing, out Thing outthing)
+
+      recipe
+        use as view0
+        use as view1
+        DoSomething as particle0
+          inthing <- view0
+          outthing -> view1
+    `));
+    let recipe = manifest.recipes[0];
+    recipe.views.forEach(v => v._originalFate = "?");
+    assert(recipe.normalize());
+    var arc = createTestArc("test-plan-arc", manifest, "dom");
+    var strategizer = {generated: [{result: manifest.recipes[0], score: 1}], terminal: []};
+
+    var ff = new FallbackFate(arc);
```

Comment:
I think it's best to avoid short/acronym names for variables. `strategy` would be oK here.

Reply:
done